tests/extras/datasets/spark/conftest.py

```diff
@@ -4,63 +4,38 @@
 discover them automatically. More info here:
 https://docs.pytest.org/en/latest/fixture.html
 """
-import gc
-from subprocess import Popen
-
 import pytest
 from delta import configure_spark_with_delta_pip
+from filelock import FileLock
 
 try:
-    from pyspark import SparkContext
     from pyspark.sql import SparkSession
 except ImportError:  # pragma: no cover
     pass  # this is only for test discovery to succeed on Python 3.8
 
-the_real_getOrCreate = None
-
-
-class UseTheSparkSessionFixtureOrMock:  # pylint: disable=too-few-public-methods
-    pass
-
-
-# prevent using spark without going through the spark_session fixture
-@pytest.fixture(scope="session", autouse=True)
-def replace_spark_default_getorcreate():
-    global the_real_getOrCreate  # pylint: disable=global-statement
-    the_real_getOrCreate = SparkSession.builder.getOrCreate
-    SparkSession.builder.getOrCreate = UseTheSparkSessionFixtureOrMock
-    return the_real_getOrCreate
 
-
-# clean up pyspark after the test module finishes
-@pytest.fixture(scope="module")
-def spark_session():  # SKIP_IF_NO_SPARK
-    SparkSession.builder.getOrCreate = the_real_getOrCreate
-    builder = (
+def _setup_spark_session():
+    return configure_spark_with_delta_pip(
         SparkSession.builder.appName("MyApp")
+        .master("local[*]")
```

Comment:
Is this telling Spark to run locally as opposed to a cluster, with however many cores it needs?

Reply:
Yea https://stackoverflow.com/questions/32356143/what-does-setmaster-local-mean-in-spark -- but it wasn't needed. I was trying it to see if that's the issue. In the beginning I thought AWS env vars cause it to confuse to run elsewhere. No harm leaving it there though